### PR TITLE
feat: Handle uppercase repo owners in actions

### PR DESCRIPTION
**Pull request by TannerGabriel**
## This PR

- Sanitizes upper-case `repository_owner` names (turns them into lower-case)
- Uses the lower-case names in all affected workflows

### Proof of work

I tried the `CI` action in my fork (`repository_owner=TannerGabriel`) and got the following results:
keptn-service-template-uppercase-repo-name

**Resolves:** #118

### CODE REVIEW

**Review by Raffy23 — APPROVED** [review]
LGTM

File: .github/workflows/initrepo.yml [hunk 1]
```diff
@@ -22,26 +22,32 @@ jobs:
...
+        with:
+          result-encoding: string
+          script: return '${{ github.repository_owner }}'.toLowerCase()
       - name: Update the image name in .ci_env, helm chart and skaffold
         run: |
           ESCAPED_ORIGINAL=$(printf '%s\n' "$ORIGINAL_IMAGE_NAME" | tr '[:upper:]' '[:lower:]' | sed -e 's/[\/&]/\\&/g')
           ESCAPED_REPLACE=$(printf '%s\n' "$NEW_IMAGE_NAME" | tr '[:upper:]' '[:lower:]' | sed -e 's/[\/&]/\\&/g') 
           sed -i "s/${ESCAPED_ORIGINAL}/${ESCAPED_REPLACE}/g" chart/values.yaml skaffold.yaml chart/README.md
-          NEW_SERVICE_NAME=$(echo ${{ github.repository }} | sed -e 's/${{ github.repository_owner }}\///')
+          NEW_SERVICE_NAME=$(echo ${{ github.repository }} | sed -e 's/${{ steps.repo_slug.outputs.result }}\///')
```
Comment (christian-kreuzberger-dtx) [inline .github/workflows/initrepo.yml:36]: Careful, this is not what you intend to do here. 

Example: `github_repository=YourName/some-service` , however `github_repository_owner=YourName`. With `sed -e ' s/...` we are trying to remove the `YourName/` part of the string, so here we don't need the lower case slug.

**Review by christian-kreuzberger-dtx — CHANGES_REQUESTED** [review]
I believe the changes in the initrepo workflow are not needed

File: .github/workflows/initrepo.yml (hunk 1 above)
Comment (TannerGabriel) [reply .github/workflows/initrepo.yml]: Sorry, I overlooked that. It should be fixed now!

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM

**Pull request merged**